trantor/net/inner/tlsprovider/BotanTLSProvider.cc

```diff
@@ -0,0 +1,390 @@
+#include <trantor/net/inner/TLSProvider.h>
+#include <trantor/net/Certificate.h>
+#include <trantor/net/TcpConnection.h>
+#include <trantor/net/callbacks.h>
+#include <trantor/utils/Logger.h>
+
+#include <botan/tls_server.h>
+#include <botan/tls_client.h>
+#include <botan/tls_callbacks.h>
+#include <botan/tls_policy.h>
+#include <botan/auto_rng.h>
+#include <botan/certstor.h>
+#include <botan/certstor_system.h>
+#include <botan/data_src.h>
+#include <botan/pkcs8.h>
+#include <botan/tls_exceptn.h>
+#include <botan/pkix_types.h>
+#include <botan/certstor_flatfile.h>
+
+using namespace trantor;
+using namespace std::placeholders;
+
+static Botan::System_RNG sessionManagerRng;
+// Is this Ok? C++ technically doesn't guarantee static object initialization
+// order.
+static Botan::TLS::Session_Manager_In_Memory sessionManager(sessionManagerRng);
+static thread_local Botan::System_RNG rng;
+static thread_local Botan::System_Certificate_Store certStore;
```

Comment:
I would guess that you don't need to make the `System_Certificate_Store` thread-local. It should be thread-safe on all platforms and by far not contended enough to need a copy per thread.